### PR TITLE
fix(wallet): Transaction Details Panel Dismissal

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Fixes bug where the `Transaction Details` panel was not dismissing on click after completing a Hardware Wallet transaction.

Resolves <https://github.com/brave/brave-browser/issues/22759>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

Before:

https://user-images.githubusercontent.com/40611140/167521958-8e0ba135-2e4d-45c7-9809-36d98bfda90d.mov

After:

https://user-images.githubusercontent.com/40611140/167521975-34a7d579-abe7-4377-b4e7-2e0d3f4baa5c.mov

### CODE REVIEW

File: components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts [hunk 1]
```diff
@@ -291,6 +292,7 @@ handler.on(PanelActions.approveHardwareTransaction.getType(), async (store: Stor
       refreshTransactionHistory(txInfo.fromAddress)
       await store.dispatch(PanelActions.setSelectedTransaction(txInfo))
       await store.dispatch(PanelActions.navigateTo('transactionDetails'))
+      apiProxy.panelHandler.setCloseOnDeactivate(true)
```
Comment (josheleonard) [inline components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts:295]: Is this handler defined twice?

File: components/brave_wallet_ui/panel/container.tsx [hunk 2]
```diff
@@ -358,15 +358,11 @@ function Container (props: Props) {
...
 
-  const retryHardwareOperation = () => {
-    // signMessageData by default initialized as [{ id: -1, address: '', message: '' }]
-    if (signMessageData && signMessageData.length && signMessageData[0].id !== -1) {
-      onSignData()
-    }
-    if (selectedPendingTransaction) {
-      onConfirmTransaction()
-    }
+  const onSelectTransaction = (transaction: BraveWallet.TransactionInfo) => {
```
Comment (josheleonard) [inline components/brave_wallet_ui/panel/container.tsx:361]: Did anything change in this file other than moving the function declaration order?

File: components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts:295]: `const apiProxy = getWalletPanelApiProxy()` is defined once
`apiProxy.panelHandler.setCloseOnDeactivate(true)` is called in for 2 separate instances 
once for `Ledger` success and once for `Trezor` success

File: components/brave_wallet_ui/panel/container.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/panel/container.tsx:361]: Just function declaration order

**Pull request merged**